File: src/main/java/org/pixeltime/enchantmentsenhance/events/enchantments/Endless.java
package org.pixeltime.enchantmentsenhance.events.enchantments;

import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.event.entity.EntityShootBowEvent;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.event.player.PlayerItemBreakEvent;
import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Endless implements Listener {
    @EventHandler(ignoreCancelled = true)
    public void noWeaponBreakDamage(final PlayerInteractEvent playerInteractEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "endless"));
        try {
            if (playerInteractEvent.getItem().getItemMeta().hasLore() && playerInteractEvent.getItem().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                playerInteractEvent.getItem().setDurability((short) 1);
            }
        } catch (Exception ex) {
        }
    }

    @EventHandler
    public void noWeaponBreakDamage(final EntityDamageByEntityEvent entityDamageByEntityEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "animal_tamer"));
        try {
            if (entityDamageByEntityEvent.getDamager() instanceof Player) {
                final Player player = (Player) entityDamageByEntityEvent.getDamager();
                if (player.getItemInHand().getItemMeta().hasLore() && player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                    ((Player) entityDamageByEntityEvent.getDamager()).getItemInHand().setDurability((short) 1);
                }
            }
            if (entityDamageByEntityEvent.getEntity() instanceof Player) {
                final Player player2 = (Player) entityDamageByEntityEvent.getEntity();
                if (player2.getInventory().getHelmet() != null) {
                    final ItemStack helmet = player2.getInventory().getHelmet();
                    if (helmet.getItemMeta().hasLore() && helmet.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                        helmet.setDurability((short) 0);
                        player2.getInventory().setHelmet(helmet);
                    }
                }
                if (player2.getInventory().getChestplate() != null) {
                    final ItemStack chestplate = player2.getInventory().getChestplate();
                    if (chestplate.getItemMeta().hasLore() && chestplate.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                        chestplate.setDurability((short) 0);
                        player2.getInventory().setChestplate(chestplate);
                    }
                }
                if (player2.getInventory().getLeggings() != null) {
                    final ItemStack leggings = player2.getInventory().getLeggings();
                    if (leggings.getItemMeta().hasLore() && leggings.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                        leggings.setDurability((short) 0);
                        player2.getInventory().setLeggings(leggings);
                    }
                }
                if (player2.getInventory().getBoots() != null) {
                    final ItemStack boots = player2.getInventory().getBoots();
                    if (boots.getItemMeta().hasLore() && boots.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                        boots.setDurability((short) 0);
                        player2.getInventory().setBoots(boots);
                    }
                }
            }
        } catch (Exception ex) {
        }
    }

    @EventHandler
    public void noWeaponBreakDamage(final EntityShootBowEvent entityShootBowEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "animal_tamer"));
        try {
            if (entityShootBowEvent.getEntity() instanceof Player && entityShootBowEvent.getBow().getItemMeta().hasLore() && entityShootBowEvent.getBow().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                entityShootBowEvent.getBow().setDurability((short) 1);
            }
        } catch (Exception ex) {
        }
    }

    @EventHandler
    public void noWeaponBreakDamage(final PlayerItemBreakEvent playerItemBreakEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "animal_tamer"));
        try {
            final ItemStack clone = playerItemBreakEvent.getBrokenItem().clone();
            if (clone.getItemMeta().hasLore() && clone.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                clone.setDurability((short) 0);
            }
            playerItemBreakEvent.getPlayer().getInventory().addItem(new ItemStack[]{clone});
        } catch (Exception ex) {
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantment/Endless.kt
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.event.enchantment

import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.entity.EntityDamageByEntityEvent
import org.bukkit.event.entity.EntityShootBowEvent
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.event.player.PlayerItemBreakEvent
import org.pixeltime.enchantmentsenhance.listener.EnchantmentListener
import org.pixeltime.enchantmentsenhance.manager.KM

class Endless : EnchantmentListener() {
    override fun desc(): Array<String> {
        return arrayOf("Item with this enchantment will never be broken", "这件物品不可被摧毁")
    }

    override fun lang(): Array<String> {
        return arrayOf("无尽")
    }

    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    fun noWeaponBreakDamage(playerInteractEvent: PlayerInteractEvent) {
        try {
            if (playerInteractEvent.item.itemMeta.hasLore() && KM.getLevel(this.name(), playerInteractEvent.item.itemMeta.lore) > 0) {
                playerInteractEvent.item.durability = 0.toShort()
            }
        } catch (ex: Exception) {
        }

    }

    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    fun noWeaponBreakDamage(entityDamageByEntityEvent: EntityDamageByEntityEvent) {
        try {
            if (entityDamageByEntityEvent.damager is Player) {
                val player = entityDamageByEntityEvent.damager as Player
                if (player.itemInHand.itemMeta.hasLore() && KM.getLevel(this.name(), player.itemInHand.itemMeta.lore) > 0) {
                    (entityDamageByEntityEvent.damager as Player).itemInHand.durability = 0.toShort()
                }
            }
            if (entityDamageByEntityEvent.entity is Player) {
                val player2 = entityDamageByEntityEvent.entity as Player
                if (player2.inventory.helmet != null) {
                    val helmet = player2.inventory.helmet
                    if (helmet.itemMeta.hasLore() && KM.getLevel(this.name(), helmet.itemMeta.lore) > 0) {
                        helmet.durability = 0.toShort()
                        player2.inventory.helmet = helmet
                    }
                }
                if (player2.inventory.chestplate != null) {
                    val chestplate = player2.inventory.chestplate
                    if (chestplate.itemMeta.hasLore() && KM.getLevel(this.name(), chestplate.itemMeta.lore) > 0) {
                        chestplate.durability = 0.toShort()
                        player2.inventory.chestplate = chestplate
                    }
                }
                if (player2.inventory.leggings != null) {
                    val leggings = player2.inventory.leggings
                    if (leggings.itemMeta.hasLore() && KM.getLevel(this.name(), leggings.itemMeta.lore) > 0) {
                        leggings.durability = 0.toShort()
                        player2.inventory.leggings = leggings
                    }
                }
                if (player2.inventory.boots != null) {
                    val boots = player2.inventory.boots
                    if (boots.itemMeta.hasLore() && KM.getLevel(this.name(), boots.itemMeta.lore) > 0) {
                        boots.durability = 0.toShort()
                        player2.inventory.boots = boots
                    }
                }
            }
        } catch (ex: Exception) {
        }

    }

    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    fun noWeaponBreakDamage(entityShootBowEvent: EntityShootBowEvent) {
        try {
            if (entityShootBowEvent.entity is Player && entityShootBowEvent.bow.itemMeta.hasLore() && KM.getLevel(this.name(), entityShootBowEvent.bow.itemMeta.lore) > 0) {
                entityShootBowEvent.bow.durability = 0.toShort()
            }
        } catch (ex: Exception) {
        }

    }

    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    fun noWeaponBreakDamage(playerItemBreakEvent: PlayerItemBreakEvent) {
        try {
            val clone = playerItemBreakEvent.brokenItem.clone()
            if (clone.itemMeta.hasLore() && KM.getLevel(this.name(), clone.itemMeta.lore) > 0) {
                clone.durability = 0.toShort()
            }
            playerItemBreakEvent.player.inventory.addItem(clone)
        } catch (ex: Exception) {
        }

    }
}
